notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-image-classification.ipynb

```diff
@@ -0,0 +1,281 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python -m transformers.onnx --model openai/clip-vit-base-patch16 clip-vit-base-patch16"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openvino.runtime import serialize\n",
+    "from openvino.tools import mo\n",
+    "\n",
+    "ov_model = mo.convert_model('clip-vit-base-patch16/model.onnx', compress_to_fp16=True)\n",
+    "serialize(ov_model, 'clip-vit-base-patch16/model.xml')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run OpenVINO model\n",
+    "\n",
+    "The steps for making predictions with OpenVINO CLIP model are the same like for PyTorch model. Let's check model result."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from scipy.special import softmax\n",
+    "from openvino.runtime import Core\n",
+    "\n",
+    "# create OpenVINO core object instance\n",
+    "core = Core()\n",
+    "# compile model for loading on device\n",
+    "compiled_model = core.compile_model(ov_model)\n",
+    "# obtain output tensor for getting predictions\n",
+    "logits_per_image_out = compiled_model.output(0)\n",
+    "# run inference on preprocessed data and get image-text similarity score\n",
+    "ov_logits_per_image = compiled_model(dict(inputs))[logits_per_image_out]\n",
+    "# perform softmax on score\n",
+    "probs = softmax(ov_logits_per_image, axis=1)\n",
+    "# visualize prediction\n",
+    "visualize_result(image, input_labels, probs[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great! Looks like we got the same result."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, is your turn. You can provide own image and comma-separated list of labels for zero-shot classification"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ipywidgets as widgets\n",
+    "style = {'description_width': 'initial'}\n",
+    "\n",
+    "image_widget = widgets.FileUpload(\n",
+    "    accept='',\n",
+    "    multiple=False,\n",
+    "    description='Upload image',\n",
+    "    style=style\n",
+    ")\n",
+    "\n",
+    "labels_widget = widgets.Textarea(\n",
+    "    value='cat,dog,bird',\n",
+    "    placeholder='Type something',\n",
+    "    description='Enter your classes separated by ,:',\n",
```

Comment:
This code should have similar structure to ```text_descriptions = [f"This is a photo of a {label}" for label in input_labels]```

Reply:
It is not clear how the reader should add the info into the code

Reply:
We need descriptions for the next two cells.

Reply:
![image](https://user-images.githubusercontent.com/10940214/208446736-d5115a19-5096-4a34-9663-277e804c3045.png)

Great result.